pkg/cli/cvd.go

```diff
@@ -155,6 +156,9 @@ func (c *cvdCreator) createCVDFromLocalBuild() ([]*hoapi.CVD, error) {
 	if err != nil {
 		return nil, fmt.Errorf("error retrieving Android Build environment variables: %w", err)
 	}
+	if err = vars.regenerateHostOut(); err != nil {
```

Comment:
the logic within method `regenerateHostOut` is about getting the target architecture and then override its `HostOut` field. Let's isolate the functionality of getting the target architecture and move it to the end of the file and modify `GetAndroidEnvVarValues`  to have it returning the relevant HostOut value.

Reply:
Just for double-checking before starting on edit, your suggestion is `GetAndroidEnvVarValues` get arch information as parameter so that we could continue to maintain our unit tests, w/o relying on any binary files like `soong_ui`, right? I'll try to work on that.

Reply:
Ah, I thought wrongly. To run `soong_ui`, it needs `ANDROID_BUILD_TOP` env. However it's extracted in `GetAndroidEnvVarValues`. So functionally,  running `soong_ui` should exist inside `GetAndroidEnvVarValues`, how should we do unit tests then? @ser-io

Reply:
I see, removed the `AndroidEnvVars` and `GetAndroidEnvVarValues` struct in https://github.com/google/cloud-android-orchestration/pull/297 to compute the values independently. After rebase, let's then create two functions: `targetArch(buildtop string)`  and `hostOut(arch string)`

Reply:
Done again, after rebasing. I think this comment is now outdated, closing this comment.